app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,29 @@
+class CommentsController < ApplicationController
+
+  def create
+    new_comment = idea_or_comment.comments.new(text: comment_params[:text])
+    new_comment.author = current_user
+    if new_comment.save
+      render partial: 'show', locals: { comment: idea_or_comment }, layout: false if idea_or_comment.is_a? Comment
```

Comment:
lots of repetitive code here - see if you can clean it up. overall, this is pretty solid though. your logic here is good though